### ISSUE TITLE
Creating environment with Shared Load Balancer, but a new ALB is also created

### ISSUE THREAD

**Issue by juanilarregui** [aws/aws-elastic-beanstalk-cli#97]
Hi, we are trying to create an environment with the EB CLI with a Shared Application Load Balancer (ALB already created in the EC2 console).

However, though we tried a bunch of different configurations, the client not only effectively uses the shared load balancer, adding the defined rules during environment creation, but it also creates a new additional ALB, that we don't require.

Here's the config file we are using for the `eb create` command, located in the `.ebextensions` subfolder:

```
option_settings:
  aws:elasticbeanstalk:application:environment:
    APP_NAME: <Here we set our app name>
  aws:ec2:instances:
    InstanceTypes: 't3.micro'
  aws:elasticbeanstalk:environment:
    LoadBalancerIsShared: true
    LoadBalancerType: application
  aws:elbv2:loadbalancer:
    SecurityGroups: <Here we set 2 security groups as sg-ID1, sg-ID2>
    SharedLoadBalancer: <ARN of the shared load balancer, already created>
  aws:ec2:vpc:
    VPCId: <Our VPC ID>
    Subnets: <Our 3 subnets in the VPC, comma-separated>
  aws:elbv2:listener:443:
    rules: default
  aws:elasticbeanstalk:environment:process:default:
    Port: '443'
    Protocol: HTTPS
```

Any clues about why we can't avoid the creation of the additional new ALB?

**Comment by Mickeypeng** [comment]
Hi juanilarregui@, thanks for reaching out. If you want to create a EB environment with existing shared application load balancer, you can use eb create and specify the `-ls` or `–shared-lb` argument.

For your use case, I assume that your full command is `eb create --cfg <configuration_number>` . In this case, the EB CLI will interactively ask about the load balancer type and whether to create a new one or use existing one. I guess that in the interactive process you chose to create a new ALB. To avoid the unnecessary creation, I would specify the `-ls` argument if performing the operation non-interactively.

P.S. See options description for eb create here: https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/eb3-create.html

**Comment by janvennemann** [comment]
@Mickeypeng I just ran into this as well and either the docs are a little misleading on this or it's a possible bug in the EB CLI. As pointed out in the docs under [Shared Application Load Balancer namespaces](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/environments-cfg-alb-shared.html#environments-cfg-alb-shared-namespaces), we should set the load balancer type to `application` and use a configuration file to configure the shared load balancer.

My expectation from reading that is that i can run `eb create my-env-dev --elb-type application` and it will pick up the shared load balancer ARN from the config file and i don't have to specify `-ls` in the CLI. However, this does not seem to be true and the EB CLI will ignore the config. Additionally, when i add the `aws:elasticbeanstalk:environment:LoadBalancerIsShared: true` setting to the config. it will begin creating the environment and indeed uses the shared load balancer from config, but will fail while trying to create a dedicated load balancer as well. This may be somewhat expected though, because the docs state that the `aws:elasticbeanstalk:environment` namespace is not supported through config files.

So long story short: The only way to create an environment with a shared load balancer via the CLI is using the `-ls` option. Setting the ARN through the config file as stated in the docs does not seem to work.

**Comment by JamesFoley1** [comment]
@juanilarregui did you ever get a resolution for this? I have been working on trying to figure this out for well over a week and it's driving me crazy! All of the config is set up as intended, except that additional load balancer that props up!

**Comment by juanilarregui** [comment]
Hi @JamesFoley1,

Unfortunately (as I remember) we couldn't find a solution and ended up taking another path. Sorry for not being able to help you!
